synapse/cortex.py

```diff
@@ -2771,6 +2812,146 @@ async def _initCoreLayers(self):
+
+        layr = self.layers.get(layriden)
+        if layr is None:
+            return
+
+        pushs = layr.layrinfo.get('pushs')
+        if pushs is None:
+            pushs = {}
+
+        pushs[iden] = pdef
+
+        await layr.layrinfo.set('pushs', pushs)
+        await self.runLayrPush(layr, pdef)
+
+    @s_nexus.Pusher.onPushAuto('layer:push:del')
+    async def delLayrPush(self, layriden, pushiden):
+
+        layr = self.layers.get(layriden)
+        if layr is None:
+            return
+
+        pushs = layr.layrinfo.get('pushs')
+        if pushs is None:
+            return
+
+        pdef = pushs.pop(pushiden, None)
+        if pdef is None:
+            return
+
+        await layr.layrinfo.set('pushs', pushs)
+        await self.delActiveCoro(pushiden)
+
+    @s_nexus.Pusher.onPushAuto('view:pull:add')
+    async def addViewPull(self, viewiden, pdef):
+
+        reqValidPull(pdef)
+
+        # TODO: schema validation
+        iden = pdef.get('iden')
+
+        view = self.views.get(viewiden)
+        if view is None:
+            return
+
+        pulls = view.info.get('pulls')
+        if pulls is None:
+            pulls = {}
+
+        pulls[iden] = pdef
+        await view.info.set('pulls', pulls)
+
+        await self.runViewPull(view, pdef)
+
+    @s_nexus.Pusher.onPushAuto('view:pull:del')
+    async def delViewPull(self, viewiden, pulliden):
+
+        view = self.views.get(viewiden)
+        if view is None:
+            return
+
+        pulls = view.info.get('pulls')
+        if pulls is None:
+            return
+
+        pdef = pulls.pop(pulliden, None)
+        if pdef is None:
+            return
+
+        await view.info.set('pulls', pulls)
+        await self.delActiveCoro(pulliden)
+
+    async def runLayrPush(self, layr, pdef):
+        url = pdef.get('url')
+        iden = pdef.get('iden')
+        # push() will refire as needed
+        async def push():
+            async with await self.boss.promote(f'layer push: {layr.iden} {iden}', self.auth.rootuser) as task:
+                async with await s_telepath.openurl(url) as proxy:
+                    await self._pushBulkEdits(layr, proxy, pdef)
+        await self.addActiveCoro(push, iden=iden)
+
+    async def runViewPull(self, view, pdef):
+        url = pdef.get('url')
+        iden = pdef.get('iden')
+        # pull() will refire as needed
+        async def pull():
+            async with await self.boss.promote(f'view pull: {view.iden} {iden}', self.auth.rootuser) as task:
+                async with await s_telepath.openurl(url) as proxy:
+                    await self._pushBulkEdits(proxy, view, pdef)
+        await self.addActiveCoro(pull, iden=iden)
+
+    async def _pushBulkEdits(self, layr0, layr1, pdef):
+
+        iden = pdef.get('iden')
+        user = pdef.get('user')
+
+        gvar = f'push:{iden}'
+
+        async with await s_base.Base.anit() as base:
+
+            queue = s_queue.Queue(maxsize=10000)
```

Comment:
there's no need to make the queue this big.

Reply:
We'll be consuming in chunks of up to 1000 queue entries at a time and i want to ensure constant/smooth consumption.  These will be nodeedits() tuples and have a very small footprint, so i see no harm in allowing up to 10k.  This approach allows large perf/efficiency gains when back-filling data ( especially into a cluster by keeping transaction counts lower ) and costs very little.